nni/compression/pytorch/speedup/v2/model_speedup.py

```diff
@@ -0,0 +1,424 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+
+from copy import deepcopy
+import logging
+from pathlib import Path
+import tempfile
+from typing import Any, Dict, List
+
+import torch
+import torch.fx
+from torch.fx import GraphModule
+from torch.fx._compatibility import compatibility
+from torch.fx.node import Node, Target
+
+from nni.common.concrete_trace_utils import concrete_trace
+from nni.compression.pytorch.utils import set_nested_attr
+from nni.compression.pytorch.speedup.compress_modules import replace_module
+from nni.compression.pytorch.utils.mask_conflict import fix_mask_conflict
+from nni.compression.pytorch.utils.utils import rand_like_with_shape, torch_integer_dtype
+
+from .container import NodeInfo
+from .mask_updater import (MaskUpdater,
+                           DefaultMaskUpdater,
+                           LeafModuleMaskUpdater,
+                           NoMaskUpdater,
+                           NoChangeMaskUpdater)
+from .replacer import Replacer, DefaultReplacer
+from .utils import tree_map_zip
+
+
+@compatibility(is_backward_compatible=True)
+class ModelSpeedup(torch.fx.Interpreter):
+    """
+    This class is to speedup the model with provided weight mask.
+
+    Parameters
+    ----------
+    model
+        The model user wants to speedup.
+    dummy_input
+        A tensor or a tuple, the dummy input to execute the model.
+    masks_file
+        The path of user provided masks file, or the masks object.
+    map_location
+        The device on which masks are placed, same to map_location in ```torch.load```.
+    batch_dim
+        The index of batch dimension in the dummy_input.
+    batch_size
+        The batch_size coefficient of the sparsity inference.
+        This value is actually used as the batchsize of the dummy_input.
+    customized_mask_updaters
+        ...
```

Comment:
what is the meaning of "..."?

Reply:
Seems a missing TODO, I will add some docstring